File: NET/JNetCall.Sharp.StdIO/Common/ByteMarks.cs
using System.IO;

namespace JNetCall.Sharp.Common
{
    public static class ByteMarks
    {
        public static (Stream stdIn, Stream stdOut) WriteSync(Stream stdOut, Stream stdIn)
        {
            const int marker = 0xEE;
            // Send flag
            stdIn.WriteByte(marker);
            stdIn.Flush();
            // Receive flag
            while (stdOut.ReadByte() != marker)
            {
            }
            // Ready!
            return (stdOut, stdIn);
        }
    }
}

File: JVM/jnetcall-java-stdio/src/main/java/jnetcall/java/common/ByteMarks.java
package jnetcall.java.common;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import org.javatuples.Pair;

import jnetproto.java.tools.Tuples;

public final class ByteMarks {

    public static Pair<InputStream, OutputStream> writeSync(InputStream stdOut, OutputStream stdIn)
            throws IOException {
        final int marker = 0xEE;
        // Send flag
        stdIn.write(marker);
        stdIn.flush();
        // Receive flag
        while (stdOut.read() != marker) ;
        // Ready!
        return Tuples.create(stdOut, stdIn);
    }
}
